src/components/VFilters/VSearchGridFilter.vue

```diff
@@ -115,13 +116,14 @@ export default defineComponent({
      * This watcher fires even when the queries are equal. We update the path only
      * when the queries change.
      */
-    watch(
+    watchDebounced(
       () => searchStore.searchQueryParams,
       (newQuery: ApiQueryParams, oldQuery: ApiQueryParams) => {
         if (!areQueriesEqual(newQuery, oldQuery)) {
           router.push(searchStore.getSearchPath())
         }
-      }
+      },
+      { debounce: 500, maxWait: 5000 }
```

Comment:
After playing with this a little, I think something like 1000 or 800 ms for the 	`debounce` value might work better and cover more cases, including situations where a user is selecting multiple filters in a row. For example, choosing multiple sources.